pandas/core/generic.py

```diff
@@ -5771,7 +5771,10 @@ def pipe(
         ...    .pipe((func, 'arg2'), arg1=a, arg3=c)
         ...  )  # doctest: +SKIP
         """
-        return common.pipe(self, func, *args, **kwargs)
+        result = common.pipe(self, func, *args, **kwargs)
+        if _using_copy_on_write():
+            return result.copy(deep=None)
```

Comment:
Should we pass in `self.copy(deep=None)` to the pipe, instead of shallow copying the result?

Or do we have guarantees at the moment about that a piped function can mutate the dataframe inplace?

Reply:
Good point, this should be the other way round, will adjust accordingly and add tests